pkg/virt-launcher/virtwrap/cli/event.go

```diff
@@ -0,0 +1,85 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package cli
+
+import (
+	"fmt"
+
+	"libvirt.org/libvirt-go"
+)
+
+type DomainEventDeviceRemovedRegistrar struct {
+	connect        *libvirt.Connect
+	domain         *libvirt.Domain
+	registrationID int
+}
+
+func NewDomainEventDeviceRemovedRegistrar(connection Connection, virDomain VirDomain) *DomainEventDeviceRemovedRegistrar {
+	libvirtConnect := extractLibvirtConnect(connection)
+	libvirtDomain := extractLibvirtDomain(virDomain)
```

Comment:
I don't see much advantage on the `extract` functions here, they are used only here and are adding one level of indirection, apart from the fact that `extartLibvirtDomain` is casting and not extracting.

Reply:
There is no "cast" in Go. It is extracting the underlying value of the type behind the interface.

The advantage is that it is easier to read and I do not see the problem with one level of indirection (which is an assumption anyway, as the compiler may inline it in the end). Are you claiming that we should not extract into functions because it has a performance penalty?
I would claim the opposite, we should always extract, until we cannot anymore or there is a real performance restraint we cannot ignore.